pymedphys/_dicom/anonymise.py

```diff
@@ -642,9 +619,50 @@ def _filter_identifying_keywords(keywords_to_leave_unchanged):
     return keywords_filtered
 
 
-def get_anonymous_replacement_value(keyword):
-    """Get an appropriate dummy anonymisation value for a DICOM element
+def get_anonymous_replacement_value(
+    keyword,
+    current_value=None,
+    replacement_strategy=anon.ANONYMISATION_HARDCODE_DISPATCH,
+):
+    """Get an appropriate anonymisation value for a DICOM element
     based on its value representation (VR)
+    Parameters
+    ----------
+    keyword: text string that is the pydicom name for the DICOM attribute/element
+
+    current_value: optional, the value that is currently assigned to the element
+
+    replacement_strategy: optional, a dispatch dictionary whose keys are the text representation
+    of the VR of the element, and whose values are function references that take the current value
+    of the element.
```

Comment:
Might be worth making reference here to where the user can import a list of keys that need to be override (eg the `KEYS_TO_OVERRIDE` defined in a previous review comment).

Do you think, if a certain key is not defined we should raise an error? Or should it fall back to the default override? I can see both pros and cons to each approach.

Reply:
they keys to the dispatch dictionary are the VR.  And there are VR in DICOM that are not in the dispatch dictionary, and that should *definitely* get checked (e.g. if vr not in dispatch_dict).  If that shows up, it's an error.  Because someone put in a keyword whose VR is not being addressed.

if you mean keywords to be replaced not including particular keywords that "we know should be included"... logging a warning seems appropriate, but not creating an error, and the log should identify the keyword but not the current value because of the potential to leak PHI in to the log.  If a user has specified a key to be ignored (not replaced), that is their prerogative.   I had to do that with PatientSex (because ANON breaks conformance for that attribute).

>import a list of keys that need to be overridden
I will have to hunt for that previous review comment.
It would be good to offer configuration for additional keywords to anonymise, but I don't want to try to do that here/yet.
I think UIDs need to be handled differently/separately, but let's have that discussion a bit later.

Reply:
I think we can keep it simple by just going, if the user provides a replacement_strategy and there is a VR on the DICOM file that is not in the strategy, then raise an error. We can leave it at that. That's a clear contract.

Failure modes can be handled elsewhere if needed.

Reply:
No stress about the list of keys. Potentially that is too hand holding.